src/sentry/models/appconnectbuilds.py

```diff
@@ -0,0 +1,54 @@
+"""Database models to keep track of the App Store Connect builds for a project.
+
+If a project enables the App Store Connect source to download dSYMs directly from Apple we
+need to keep track of which builds have already been downloaded.
+"""
+
+from django.db import models
+
+from sentry.db.models import FlexibleForeignKey, Model
+
+
+class AppConnectBuild(Model):
+    """A single build that exists or has existed on App Store Connect.
+
+    Builds on App Store Connect expire, so they do disappear from the API.  This table does
+    not get pruned however, once we know of a build we know of it forever.
+
+    An individual build can be identified by the tuple of (bundle_id, platform,
+    bundle_short_version, bundle_version).
+    """
+
+    __core__ = True
+
+    project = FlexibleForeignKey("sentry.Project")
```

Comment:
It's slightly risky to add a db level constraint to `Project`, so generally I recommend using `db_constraint=False` here.

Reply:
done